signer/src/transaction_coordinator.rs

```diff
@@ -2540,6 +2540,21 @@ pub async fn should_coordinate_dkg(
     let storage = context.get_storage();
     let config = context.config();
 
+    // Trigger dkg if signer set changes
+    let last_dkg_signer_set: BTreeSet<_> = context
+        .state()
+        .current_signer_set()
+        .get_signers()
+        .iter()
+        .map(|signer| *signer.public_key())
+        .collect();
+    let config_signer_set = config.signer.bootstrap_signing_set();
```

Comment:
Maybe we could consolidate this logic into the state since it's done in multiple places? e.g. something like:
```rust
let changed = !context
    .state()
    .current_signer_set()
    .is_same(&config.signer.bootstrap_signing_set());
```

Reply:
[cd9d6cd](https://github.com/stacks-sbtc/sbtc/pull/1602/commits/cd9d6cd531acee663b5cb3a3a387d632ddc68f0e)

Reply:
`have` is the wrong form here, could we name it `has_same_pubkeys` (or `cmp_pubkeys`) instead?

Reply:
Whooops, English......

About `cmp_pubkeys` imo it is less obvious how to interpret it's results.

Reply:
[e9bdafb](https://github.com/stacks-sbtc/sbtc/pull/1602/commits/e9bdafb256bc8649d96d08a6285911bdd4ebc6f4)